app/scripts/controllers/push-platform-notifications/utils/get-notification-message.ts

```diff
@@ -67,7 +69,12 @@ export async function onNotificationClick(event: NotificationEvent) {
   const data: OnChainRawNotification = event?.notification?.data;
 
   // Navigate
-  const destination = `chrome-extension://${sw.location.host}/home.html#notifications/${data.id}`;
+  const getPlatformResult = getPlatform();
```

Comment:
Double check this `getPlatform` code.
This calls `window.navigator`, and the `window` does not exist on service workers.
We might want to replace this with `global.navigator` or `self.navigator`

@danjm is this function `getPlatform` used in other controllers? We are not seeing any regressions or errors using this? If it works, then feel free to keep using it @matteoscurati .

Reply:
Because of this: https://github.com/MetaMask/metamask-extension/blob/develop/app/scripts/init-globals.js#L17

`window`, in our `app/` code under mv3, refers to the `ServiceWorkerGlobalScope` and not an actual `window`. The `ServiceWorkerGlobalScope` has some, but not all, of the apis an actual `window` has. `navigator` is one of the APIs it does have, so this should be fine